docs/configuration/single-process-config-blocks-local.yaml

```diff
@@ -17,6 +17,9 @@ server:
 
 distributor:
   shard_by_all_labels: true
+  ring:
```

Comment:
The change here is to solve the integration test error of TestConfigAPIEndpoint. I still haven't figured out why the changes here are caused, and I need your help to confirm that there are no potential problems with the changes here. @yeya24 @friedrichg 

```
=== RUN   TestConfigAPIEndpoint
14:38:31 Starting cortex-1
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.275326352Z caller=mapper.go:50 msg="unable to read rules directory" path=/rules err="open /rules: no such file or directory"
14:38:32 cortex-1: level=warn ts=2022-12-06T14:38:32.276273962Z caller=modules.go:320 msg="Worker address is empty in single binary mode.  Attempting automatic worker configuration.  If queries are unresponsive consider configuring the worker explicitly." address=127.0.0.1:9095
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.278520288Z caller=cortex.go:434 msg="module failed" module=distributor-service err="invalid service state: Failed, expected: Running, failure: not healthy, 0 terminated, 1 failed: [unable to initialise ring state: Get \"http://localhost:8500/v1/kv/collectors/distributor?stale=\": dial tcp 127.0.0.1:8500: connect: connection refused]"
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.27871809Z caller=client.go:155 msg="error getting key" key=collectors/distributor err="Get \"http://localhost:8500/v1/kv/collectors/distributor\": dial tcp 127.0.0.1:8500: connect: connection refused"
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.278999793Z caller=cortex.go:434 msg="module failed" module=querier err="failed to start querier, because it depends on module distributor-service, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: not healthy, 0 terminated, 1 failed: [unable to initialise ring state: Get \"http://localhost:8500/v1/kv/collectors/distributor?stale=\": dial tcp 127.0.0.1:8500: connect: connection refused]"
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.279025893Z caller=cortex.go:434 msg="module failed" module=ruler err="failed to start ruler, because it depends on module distributor-service, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: not healthy, 0 terminated, 1 failed: [unable to initialise ring state: Get \"http://localhost:8500/v1/kv/collectors/distributor?stale=\": dial tcp 127.0.0.1:8500: connect: connection refused]"
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.279360097Z caller=client.go:155 msg="error getting key" key=collectors/distributor err="Get \"http://localhost:8500/v1/kv/collectors/distributor\": dial tcp 127.0.0.1:8500: connect: connection refused"
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.2796179Z caller=client.go:155 msg="error getting key" key=collectors/distributor err="Get \"http://localhost:8500/v1/kv/collectors/distributor\": dial tcp 127.0.0.1:8500: connect: connection refused"
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.279867703Z caller=client.go:155 msg="error getting key" key=collectors/distributor err="Get \"http://localhost:8500/v1/kv/collectors/distributor\": dial tcp 127.0.0.1:8500: connect: connection refused"
14:38:32 cortex-1: level=error ts=2022-12-06T14:38:32.279880403Z caller=log.go:117 msg="error running cortex" err="failed services\ngithub.com/cortexproject/cortex/pkg/cortex.(*Cortex).Run\n\t/__w/cortex/cortex/pkg/cortex/cortex.go:469\nmain.main\n\t/__w/cortex/cortex/cmd/cortex/main.go:196\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1594"
    api_endpoints_test.go:51: 
        	Error Trace:	/home/runner/work/cortex/cortex/integration/api_endpoints_test.go:51
        	Error:      	Received unexpected error:
        	            	docker container cortex-1 failed to start: exit status 1
        	Test:       	TestConfigAPIEndpoint
--- FAIL: TestConfigAPIEndpoint (23.31s)
```

Reply:
Is this change necessary for this pr? If not, I recommend a separate pr for changing this. Looks like the failure relates to this change so let's make our life easier.

Reply:
> Is this change necessary for this pr? If not, I recommend a separate pr for changing this. Looks like the failure relates to this change so let's make our life easier.

Emm, I mean if there is no change here, this error will appear.

Reply:
consul is the default ring distributor provider.
The example worked because consul was never queried when local strategy is used. 

your PR is currently making the distributor query consul when using local strategy. That is not correct.

Reply:
It is fine to have error in CI if it is a flaky test. We can still merge your code if the error is not related to your change.
The flaky test should be a separate task to fix.